frontend/src/logic/tasklogic.ts

```diff
@@ -1,287 +1,317 @@
-import { db } from "../index";
-import { Status } from "../types";
-import { ensureNonEmpty, ensureNonNegative, isEmptyString, shouldShowExpired, sortByReverseOrder } from "./utilities";
-
-const COLLECTION_TASKS: string = "tasks";
+import { db } from '../index'
+import { Status } from '../types'
+import {
+    ensureNonEmpty,
+    ensureNonNegative,
+    isEmptyString,
+    shouldShowExpired,
+    sortByReverseOrder,
+} from './utilities'
+
+const COLLECTION_TASKS: string = 'tasks'
 
 export class Task {
-  id: string;
-  shopLocation: string;
-  expectedDeliveryTime: number;
-  item: string;
-  payerUid: string;
-  fee: number;
-  status: Status;
-  uid?: string; // equivalent to doer's id
-
-  /**
-   * @throws Error if `id`, `shopLocation`, `shopLocation`, `expectedDeliveryTime`, `item`, `payerUid` or `status` are empty.
-   * @throws Error if `fee` is not non-negative.
-   */
-  constructor(
-    id: string,
-    shopLocation: string,
-    expectedDeliveryTime: number,
-    item: string,
-    payerUid: string,
-    fee: number,
-    status: Status,
-    uid?: string
-  ) {
-    try {
-      ensureNonEmpty(
-        id,
-        shopLocation,
-        expectedDeliveryTime,
-        item,
-        payerUid,
-        fee,
-        status
-      );
-      ensureNonNegative(fee);
-    } catch (e) {
-      throw new Error(`Unable to create task: e.message`);
-    }
-
-    this.id = id;
-    this.shopLocation = shopLocation;
-    this.expectedDeliveryTime = expectedDeliveryTime;
-    this.item = item;
-    this.payerUid = payerUid;
-    this.fee = fee;
-    this.status = status;
-
+    id: string
+    shopLocation: string
+    expectedDeliveryTime: number
+    item: string
+    payerUid: string
+    fee: number
+    status: Status
+    uid?: string // equivalent to doer's id
+
+    /**
+     * @throws Error if `id`, `shopLocation`, `shopLocation`, `expectedDeliveryTime`, `item`, `payerUid` or `status` are empty.
+     * @throws Error if `fee` is not non-negative.
+     */
+    constructor(
+        id: string,
+        shopLocation: string,
+        expectedDeliveryTime: number,
+        item: string,
+        payerUid: string,
+        fee: number,
+        status: Status,
+        uid?: string
+    ) {
+        try {
+            ensureNonEmpty(
+                id,
+                shopLocation,
+                expectedDeliveryTime,
+                item,
+                payerUid,
+                fee,
+                status
+            )
+            ensureNonNegative(fee)
+        } catch (e) {
+            throw new Error(`Unable to create task: e.message`)
+        }
+
+        this.id = id
+        this.shopLocation = shopLocation
+        this.expectedDeliveryTime = expectedDeliveryTime
+        this.item = item
+        this.payerUid = payerUid
+        this.fee = fee
+        this.status = status
+        /*
     if (!Task.isValidState(uid, status)) {
       throw new Error("Missing doer for task with id " + id);
-    }
+    }*/
```

Comment:
@pohlinwei This was causing issue when i was fetching on marketplace. can check ?